app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/alwayson/AlwaysOnAlertDialogFragment.kt

```diff
@@ -0,0 +1,110 @@
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.di.scopes.FragmentScope
+import com.duckduckgo.mobile.android.ui.store.AppTheme
+import com.duckduckgo.mobile.android.vpn.R
+import com.duckduckgo.mobile.android.vpn.databinding.ContentVpnAlwaysOnAlertBinding
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dagger.android.support.AndroidSupportInjection
+import java.lang.ref.WeakReference
+import javax.inject.Inject
+
+@InjectWith(FragmentScope::class)
+class AlwaysOnAlertDialogFragment private constructor() : BottomSheetDialogFragment() {
+
+    @Inject lateinit var appBuildConfig: AppBuildConfig
+    @Inject lateinit var appTheme: AppTheme
+    private var listener: WeakReference<Listener> = WeakReference(null)
+
+    override fun getTheme(): Int = R.style.AlwaysOnBottomSheetDialogTheme
+
+    override fun onAttach(context: Context) {
+        AndroidSupportInjection.inject(this)
+        super.onAttach(context)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return ContentVpnAlwaysOnAlertBinding.inflate(inflater, container, false).apply {
+            configureViews(this)
+        }.root
+    }
+
+    private fun configureViews(binding: ContentVpnAlwaysOnAlertBinding) {
+        fun animatedClosed() {
+            (dialog as BottomSheetDialog).behavior.state = BottomSheetBehavior.STATE_HIDDEN
+        }
+
+        fun configureCloseButtons(binding: ContentVpnAlwaysOnAlertBinding) {
+            binding.closeButton.setOnClickListener { animatedClosed() }
+            binding.notNowButton.setOnClickListener { animatedClosed() }
+            binding.goToSettingsButton.setOnClickListener {
+                animatedClosed()
+                listener.get()?.onGoToSettingsClicked()
+            }
+        }
+
+        fun configurePromotionIllustration(binding: ContentVpnAlwaysOnAlertBinding) {
+            binding.alwaysOnIllustration.setAnimation(
+                if (appTheme.isLightModeEnabled()) {
+                    R.raw.always_on
+                } else {
+                    R.raw.always_on_dark
+                }
+            )
+        }
+
+        fun configureBehavior() {
+            (dialog as BottomSheetDialog).behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        }
+
+        fun bindViewElements() {
+            binding.alwaysOnModalDescription.text = HtmlCompat.fromHtml(getString(R.string.atp_AlwaysOnModalBody), 0)
+        }
+
+        configureBehavior()
+        configureCloseButtons(binding)
+        configurePromotionIllustration(binding)
+        bindViewElements()
+    }
+
+    fun interface Listener {
+        /**
+         * Called when the user clicks on the "Go to settings" button.
+         */
+        fun onGoToSettingsClicked()
```

Comment:
Only exposing this method in the listener because we really don't need anything else.